From fd27ceb5e3a8b9c9106788992112418959ba61e8 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 23 Nov 2023 16:29:37 +0100
Subject: [PATCH 0365/1695] dts: stm32: activate tamper node for STM32MP257F-DK
 board

Add tampers configuration for this board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 29a7a63dd8337518cb02456ed97fafd7ee02d68e dts: stm32: activate tamper node for STM32MP257F-DK board]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I5ebb5ad06bc55879f744e24cada50de63a081389
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343128
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349412
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 0233ca5b9..903714425 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 
 #include <dt-bindings/soc/stm32mp-provisioning.h>
+#include <dt-bindings/tamper/st,stm32mp25-tamp.h>
 #include "stm32mp257.dtsi"
 #include "stm32mp25xf.dtsi"
 #include "stm32mp25-pinctrl.dtsi"
@@ -50,6 +51,20 @@
 	status = "okay";
 };
 
+&tamp {
+	status = "okay";
+	st,tamp-passive-precharge = <2>;
+	st,tamp-passive-nb-sample = <4>;
+	st,tamp-passive-sample-clk-div = <16384>;
+
+	/* Tamper button */
+	tamp_button: tamp-passive@3 {
+		status = "disabled";
+		reg = <EXT_TAMPER_3 0>;
+		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+	};
+};
+
 &usart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.47.2

